### PR TITLE
Absorb backend

### PR DESCRIPTION
**Pull request by nyarly**
This is a fairly huge PR: as much as should be in common between Xing backend apps has been extracted from Xing Application Base. Configuration has been left in place there, so that it serves as a starting point, but that any changes to the the Xing backend should be unique to the particular project.

For comparison, of what this mean in app base, see: XingFramework/xing-application-base@19fbc8493261659e4edb918141d77b85d290bfb3

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Ability
```
Comment (hannahhoward) [inline app/models/ability.rb:1]: regarding ability.rb moving out — i get that people can customize abilities by making roles, but don’t we want them to have full control over abilities if they want it? so maybe it’s worth putting ability.rb back in app-base

File: db/migrate/20140828011806_initial.rb [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	  enable_extension "hstore"
+
+	  create_table "content_blocks", force: true do |t|
+	    t.string   "content_type"
+	    t.text     "body"
+	    t.datetime "created_at"
+	    t.datetime "updated_at"
+	  end
+
+	  create_table "documents", force: true do |t|
```
Comment (hannahhoward) [inline db/migrate/20140828011806_initial.rb]: Hmm... I'm not sure CMS migrations should live here.

File: lib/xing/nominal/database_config_validator.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
+require 'xing/nominal/yaml_config_validator'
```
Comment (hannahhoward) [inline lib/xing/nominal/database_config_validator.rb:1]: Why Nominal? Just curious. Or should all of this be in under services/nominal ?

File: lib/xing/snapshot.rb [hunk 4]
```diff
@@ -0,0 +1,4 @@
+require 'xing/snapshot/local_site_snapshot'
```
Comment (hannahhoward) [inline lib/xing/snapshot.rb:1]: I agree that snapshot should be consolidated, but should it be under services/snapshot? I'm not sure, just asking.

File: lib/xing/tasks/generate_dummy_api.rake [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  task :wipe_dummy_directory do
+    sh "rm -rf #{TARGET_DIRECTORY}"
+  end
+
+  desc "Generate API fixtures in /dummy-api from development mode endpoints"
+  task :generate => :environment do
+    require 'fileutils'
+
+    endpoints = FIXED_ENDPOINTS
+    puts "Making public pages"
```
Comment (hannahhoward) [inline lib/xing/tasks/generate_dummy_api.rake]: This seems CMS Specific.

File: lib/xing/tasks/generate_dummy_api.rake [hunk 6]
```diff
@@ -0,0 +1,56 @@
+namespace :dummy_api do
```
Comment (hannahhoward) [inline lib/xing/tasks/generate_dummy_api.rake]: Is this task one we're still using? Should it be deleted?

File: lib/xing/tasks/generate_dummy_api.rake (hunk 5 above)
Comment (nyarly) [inline lib/xing/tasks/generate_dummy_api.rake]: The FIXED_ENDPOINTS thing is related to snapshotting - likewise a target for later extraction

File: lib/xing/nominal/database_config_validator.rb (hunk 3 above)
Comment (nyarly) [inline lib/xing/nominal/database_config_validator.rb:1]: IMO services should be utils for the larger Rails app. nominal and snapshot are both support which should be extracted to own gems - separating their files will make that easier.

**Pull request merged**